back/courses/views.py

```diff
@@ -126,3 +139,27 @@ def update_timeslots(request):
             context["date_updated"] = date
 
     return render(request, "courses/update_timeslots.html", context)
+
+
+@login_required
+def join_group(request, group_id, action):
+    group = get_object_or_404(Group, id=group_id)
+    student = get_object_or_404(Student, user__id=request.user.id)
+    for group_to_delete in group.course.groups.all():
```

Comment:
Perhaps we should check if the user is not trying to join a group he/she is already into, in order to avoid useless database operations

Reply:
This shouldn't happened a priori, the button should be on 'Leave' if the student is already on a group. However I could check if he is already on the NULL group, but I think it would loose readability for only a small gain in database operations. What do you think ?

NB : I changed the loop to remove only the groups of the course that the student follow.